react-client-app/src/components/DatasetAdminForm.jsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useEffect, useRef } from 'react'
+import {
+    Select, Box, Text, Button, Flex, Spinner, Container, Input, Tooltip, CloseButton, ScaleFade, useDisclosure, Switch,
+    FormControl, FormLabel, FormErrorMessage
+} from "@chakra-ui/react"
+import PageHeading from './PageHeading'
+import ToastAlert from './ToastAlert'
+import ConceptTag from './ConceptTag'
+import { useGet, usePatch, useDelete } from '../api/values'
+
+const DatasetAdminForm = ({ setTitle }) => {
+    let datasetId = window.location.pathname.split("/").pop()
+    const { isOpen, onOpen, onClose } = useDisclosure()
+
+    // Set up component state
+    const [alert, setAlert] = useState({ hidden: true, title: '', description: '', status: 'error' })
+    const [dataset, setDataset] = useState({})
+    const [dataPartners, setDataPartners] = useState();
+    const [selectedDataPartner, setSelectedDataPartner] = useState()
+    const [isPublic, setIsPublic] = useState()
+    const [loadingMessage, setLoadingMessage] = useState("Loading page")
+    const [formErrors, setFormErrors] = useState({})
+    const [uploadLoading, setUploadLoading] = useState(false)
+    const [viewers, setViewers] = useState([])
+    const [admins, setAdmins] = useState([])
+    const [usersList, setUsersList] = useState(undefined)
+
+    function getUsersFromIds(userIds, userObjects) {
+        /**
+         * Get an array user objects with ids in an array of ids.
+         * 
+         * userIds: Array[Number]
+         * userObjects: Array[Object]
+         */
+        const idIterator = userIds.values()
+        let users = []
+        for (let id of idIterator) {
+            for (let obj of userObjects) {
+                if (id === obj.id) {
+                    users.push(obj)
+                }
+            }
+        }
+        return users
+    }
+
+    // Set up page
+    useEffect(
+        async () => {
+            setTitle(null)
+            // Get dataset
+            const datasetQuery = await useGet(`/datasets/${datasetId}`)
+            setDataset(datasetQuery)
+            setIsPublic(datasetQuery.visibility === "PUBLIC")
+            const dataPartnerQuery = await useGet("/datapartners/")
```

Comment:
Could speed this up by running them all in parallel in a promise and using Promise.all() to get the results, if the queries aren't reliant on eachother

Reply:
@Samad198 is that different to just putting `await` in front of lines 53 and 54?